lib/drivers/abbottFreeStyleLite.js

```diff
@@ -0,0 +1,538 @@
+
+var TZOUtil = require('../TimezoneOffsetUtil');
+
+var debug = require('../bows')('FreeStyleLiteDriver');
+
+module.exports = function (config) {
+  var cfg = _.clone(config);
+  cfg.deviceData = null;
+  var serialDevice = config.deviceComms;
+
+  // With no date & time settings changes available,
+  // timezone is applied across-the-board
+  cfg.tzoUtil = new TZOUtil(cfg.timezone, new Date().toISOString(), []);
+
+  var STX = 0x02;
+  var ETX = 0x03;
+  var CR = 0x0D;
+  var LF = 0x0A;
+
+  var dateTimePattern = '([A-Za-z ]{4} [0-3][0-9] [0-9]{4} ' +
+                        '[012][0-9]:[0-5][0-9](?::[0-6][0-9])?)';
+  function parseDateTime(match, index) {
+    var fmt = 'MMMM DD YYYY HH:mm:ss';
+    var parsed = sundial.parseFromFormat(match[index], fmt);
+    var dev = sundial.formatDeviceTime(parsed);
+    var utcInfo = cfg.tzoUtil.lookup(parsed);
+    return {
+      dev: dev,
+      utc: utcInfo.time,
+      timezoneOffset: utcInfo.timezoneOffset,
+      conversionOffset: utcInfo.conversionOffset
+    };
+  }
+
+  var patterns = {
+    serialNumber: {
+      pattern: '^(?:[A-Z]-....-.....|[A-Z][A-Z][A-Z]....-.....|[XC][A-Z][GM]........)$',
+      handler: function(match, packet) {
+        packet.serialNumber = match[0];
+      }
+    },
+    softwareVersion: {
+      pattern: '^[0-9.]{4,13}',
+      handler: function(match, packet) {
+        packet.softwareVersion = match[0];
+      }
+    },
+    deviceTime: {
+      pattern: '^' + dateTimePattern + '$',
+      handler: function(match, packet) {
+        var times = parseDateTime(match, 1);
+        packet.deviceTime = times.dev;
+        packet.time = times.utc;
+      }
+    },
+    numEntries: {
+      pattern: '^(?:[0-9]{3}|Log Empty)$',
+      handler: function(match, packet) {
+        packet.numEntries = parseInt(match[0], 10);
+        if (_.isNaN(packet.numEntries)) {
+          packet.numEntries = 0;
+        }
+      }
+    },
+    notFound: {
+      pattern: '^Log Not Found$',
+      handler: function(match, packet) {
+        packet.notFound = true;
+      }
+    },
+    end: {
+      pattern: '^END$',
+      handler: function(match, packet) {
+        packet.end = true;
+      }
+    },
+    ok: {
+      pattern: '^CMD OK$',
+      handler: function(match, packet) {
+        packet.ok = true;
+      }
+    },
+    fail: {
+      pattern: '^CMD Fail!$',
+      handler: function(match, packet) {
+        packet.fail = true;
+      }
+    },
+    logEntry: {
+      pattern: '^\s*([0-9HL][0-9OI][0-9 ])  ' + dateTimePattern + ' [0-9A-Fa-f][0-9A-Fa-f] (0x[0-9A-Fa-f][0-9A-Fa-f])',
+      handler: function(match, packet) {
+        var entry = {};
+        var reading = match[1];
+
+        var testType = match[3];
+
+        if(testType[3] === '1') {
+          debug('Marking as control test');
+          entry.control = true;
+        }else{
+          entry.control = false;
+        }
+
+        //this works as a reference, from the standard HI/LO values
```

Comment:
This was a comment from IOET. I've just checked the spec again - it looks like if the HI/LO flag is set, the actual value sent will be the threshold, e.g. for a reading of 567 (with a range of 20-500), the returned value will be 500 with the HI flag set.  I think they assumed that it will always be 20-500, but I don't think this will always be the case. Will look into it.